File: web/src/utils/Theme.ts
import { getEmbeddedVariable } from "./Configuration";

export function useTheme() {
    return getEmbeddedVariable("theme-name");
}

export function usePrimaryColor() {
    return getEmbeddedVariable("theme-primarycolor");
}

export function useSecondaryColor() {
    return getEmbeddedVariable("theme-secondarycolor");
}

File: internal/configuration/validator/theme.go
package validator

import (
	"github.com/authelia/authelia/internal/configuration/schema"
)

// ValidateTOTP validates and update TOTP configuration.
func ValidateTheme(configuration *schema.ThemeConfiguration, validator *schema.StructValidator) {
	if configuration.Name == "" {
		configuration.Name = schema.DefaultThemeConfiguration.Name
	}

	if configuration.PrimaryColor == "" {
		configuration.PrimaryColor = schema.DefaultThemeConfiguration.PrimaryColor
	}

	if configuration.SecondaryColor == "" {
		configuration.SecondaryColor = schema.DefaultThemeConfiguration.SecondaryColor
	}
}
